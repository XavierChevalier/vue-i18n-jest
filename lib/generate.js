"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const json5_1 = __importDefault(require("json5"));
const js_yaml_1 = __importDefault(require("js-yaml"));
const debug_1 = require("debug");
// constants
exports.VUE_OPTIONS = '__vue__options__';
exports.VUE_I18N_OPTION = '__i18n';
const defaultLang = 'json';
const debug = debug_1.debug('vue-i18n-jest');
function generate(blocks) {
    const base = `${exports.VUE_OPTIONS}.${exports.VUE_I18N_OPTION} = []`;
    const codes = blocks.map(block => {
        if (block.type === 'i18n') {
            const lang = (block.attrs && block.attrs.lang) || defaultLang;
            // const lang = block.attrs?lang ?? defaultLang
            const data = convert(block.content, lang);
            const value = JSON.stringify(JSON.parse(data))
                .replace(/\u2028/g, '\\u2028')
                .replace(/\u2029/g, '\\u2029')
                .replace(/\\/g, '\\\\');
            return `${exports.VUE_OPTIONS}.${exports.VUE_I18N_OPTION}.push('${value.replace(/\u0027/g, '\\u0027')}')`;
        }
        else {
            return '';
        }
    });
    const code = [base].concat(codes).join('\n');
    debug('genrateCode', code);
    return code;
}
exports.default = generate;
function convert(source, lang) {
    switch (lang) {
        case 'yaml':
        case 'yml':
            return JSON.stringify(js_yaml_1.default.safeLoad(source), undefined, '\t');
        case 'json5':
            return JSON.stringify(json5_1.default.parse(source));
        default:
            return source;
    }
}
