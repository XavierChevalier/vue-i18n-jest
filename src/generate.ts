import JSON5 from 'json5'
import yaml from 'js-yaml'
import { debug as Debug } from 'debug'

// type importing
import { SFCBlock } from 'vue-template-compiler'

// constants
export const VUE_OPTIONS = '__vue__options__'
export const VUE_I18N_OPTION = '__i18n'
const defaultLang = 'json'

const debug = Debug('vue-i18n-jest')

export default function generate (blocks: SFCBlock[]) {
  const base = `${VUE_OPTIONS}.${VUE_I18N_OPTION} = []`
  const codes = blocks.map(block => {
    if (block.type === 'i18n') {
      const lang = (block.attrs && block.attrs.lang) || defaultLang
      // const lang = block.attrs?lang ?? defaultLang
      const data = convert(block.content, lang)
      const value = JSON.stringify(JSON.parse(data))
        .replace(/\u2028/g, '\\u2028')
        .replace(/\u2029/g, '\\u2029')
        .replace(/\\/g, '\\\\')
      return `${VUE_OPTIONS}.${VUE_I18N_OPTION}.push('${value.replace(/\u0027/g, '\\u0027')}')`
    } else {
      return ''
    }
  })

  const code = [base].concat(codes).join('\n')
  debug('genrateCode', code)
  return code
}

function convert (source: string, lang: string): string {
  switch (lang) {
    case 'yaml':
    case 'yml':
      return JSON.stringify(yaml.safeLoad(source), undefined, '\t')
    case 'json5':
      return JSON.stringify(JSON5.parse(source))
    default:
      return source
  }
}
