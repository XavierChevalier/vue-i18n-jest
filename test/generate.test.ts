import * as fs from 'fs'
import * as path from 'path'
import { parseComponent } from 'vue-template-compiler'
import generate from '../src/generate'

const files = [
  'default.vue',
  'json.vue',
  'yaml.vue',
  'json5.vue',
  'json-locale-import.vue',
  'json5-locale-import.vue',
  'yaml-locale-import.vue'
]
const targets = files.map(file => {
  const sourcePath = path.resolve(__dirname, `./fixtures/${file}`)

  return {
    content: fs.readFileSync(sourcePath).toString(),
    sourcePath,
    file
  }
})

test('generate', () => {
  targets.forEach(target => {
    const { customBlocks } = parseComponent(target.content, { pad: true })
    const code = generate(customBlocks, target.sourcePath)
    expect(code).toMatchSnapshot(target.file)
  })
})
