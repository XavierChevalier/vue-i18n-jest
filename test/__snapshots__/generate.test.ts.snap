// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate: default.vue 1`] = `
"__vue__options__.__i18n = []
__vue__options__.__i18n.push('{\\"ja\\":{\\"hello\\":\\"こんにちは！\\"},\\"en\\":{\\"hello\\":\\"hello!\\"}}')"
`;

exports[`generate: json.vue 1`] = `
"__vue__options__.__i18n = []
__vue__options__.__i18n.push('{\\"ja\\":{\\"hello\\":\\"こんにちは！\\"},\\"en\\":{\\"hello\\":\\"hello!\\"}}')"
`;

exports[`generate: json-locale-import.vue 1`] = `
"__vue__options__.__i18n = []
__vue__options__.__i18n.push('{\\"ja\\":{\\"hello\\":\\"こんにちは！\\"},\\"en\\":{\\"hello\\":\\"hello!\\"}}')"
`;

exports[`generate: json5.vue 1`] = `
"__vue__options__.__i18n = []
__vue__options__.__i18n.push('{\\"ja\\":{\\"hello\\":\\"こんにちは！\\"},\\"en\\":{\\"hello\\":\\"hello!\\"}}')"
`;

exports[`generate: json5-locale-import.vue 1`] = `
"__vue__options__.__i18n = []
__vue__options__.__i18n.push('{\\"ja\\":{\\"hello\\":\\"こんにちは！\\"},\\"en\\":{\\"hello\\":\\"hello!\\"}}')"
`;

exports[`generate: yaml.vue 1`] = `
"__vue__options__.__i18n = []
__vue__options__.__i18n.push('{\\"ja\\":{\\"hello\\":\\"こんにちは！\\"},\\"en\\":{\\"hello\\":\\"hello!\\"}}')"
`;

exports[`generate: yaml-locale-import.vue 1`] = `
"__vue__options__.__i18n = []
__vue__options__.__i18n.push('{\\"ja\\":{\\"hello\\":\\"こんにちは！\\"},\\"en\\":{\\"hello\\":\\"hello!\\"}}')"
`;
